src/Common/src/System/Security/Cryptography/CngPkcs8.cs

```diff
@@ -0,0 +1,336 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.Asn1;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class CngPkcs8
+    {
+        // Windows 7, 8, and 8.1 don't support PBES2 export, so use
+        // the 3DES-192 scheme from PKCS12-PBE whenever deferring to the system.
+        private static readonly PbeParameters s_platformParameters =
+            new PbeParameters(
+                PbeEncryptionAlgorithm.TripleDes3KeyPkcs12,
+                HashAlgorithmName.SHA1,
+                10000);
+
+        internal static bool IsPlatformScheme(PbeParameters pbeParameters)
+        {
+            Debug.Assert(pbeParameters != null);
+
+            return pbeParameters.EncryptionAlgorithm == s_platformParameters.EncryptionAlgorithm &&
+                   pbeParameters.HashAlgorithm == s_platformParameters.HashAlgorithm;
+        }
+
+        internal static byte[] ExportEncryptedPkcs8PrivateKey(
+            AsymmetricAlgorithm key,
+            ReadOnlySpan<byte> passwordBytes,
+            PbeParameters pbeParameters)
+        {
+            if (pbeParameters == null)
+            {
+                throw new ArgumentNullException(nameof(pbeParameters));
+            }
+
+            PasswordBasedEncryption.ValidatePbeParameters(
+                pbeParameters,
+                ReadOnlySpan<char>.Empty,
+                passwordBytes);
+
+            if (passwordBytes.Length == 0)
+            {
+                // Switch to character-based, since that's the native input format.
+                return key.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, pbeParameters);
+            }
+
+            AsnWriter writer = RewriteEncryptedPkcs8PrivateKey(
+                key,
+                passwordBytes,
+                pbeParameters);
+
+            using (writer)
```

Comment:
Nit: could just be:
```C#
using (AsnWriter writer = RewriteEncryptedPkcs8PrivateKey(key, passwordBytes, pbeParameters)
```

Reply:
Hmm.  I must have written it when the data triplet that became PbeParameters was still three adjacent parameters.